src/common/harvest/harvest.js

```diff
@@ -239,9 +239,8 @@ export class Harvest extends SharedContext {
    */
   cleanPayload (payload = {}) {
     const clean = (input) => {
-      if ((typeof Uint8Array !== 'undefined' && input instanceof Uint8Array) || typeof input === 'string') {
-        return input.length > 0 ? input : null
-      }
+      if ((typeof Uint8Array !== 'undefined' && input instanceof Uint8Array) || Array.isArray(input)) return input
+      if (typeof input === 'string') return input.length > 0 ? input : null
```

Comment:
Was there a case where `input.length > 0` was not working for the other data types? Just wondering why string got split to its own line.

Reply:
I forget at this point but I think it was because returning null for the array if 0 length was breaking something. We could look deeper

Reply:
This is fine. If `input` is an empty array, it would return `null` which gets stringified to `'null'` in the `_send` method bypassing the empty check on line 120.